pennylane/tape/jacobian_tape.py

```diff
@@ -730,11 +731,19 @@ def hessian(self, device, params=None, **options):
 
             if hessian is None:
                 # create the Hessian matrix
-                hessian = np.zeros((len(params), len(params)), dtype=float)
+                if self.output_dim is not None:
+                    hessian = np.zeros(
+                        (len(params), len(params), np.prod(self.output_dim)), dtype=float
+                    )
+                else:
+                    hessian = np.zeros((len(params), len(params)), dtype=float)
```

Comment:
Maybe for consistency, we could add a dummy dimension of length 1.

```suggestion
                    hessian = np.zeros((len(params), len(params), 1), dtype=float)
```